CMake/telesculptor-external-fletch.cmake

```diff
@@ -23,18 +51,18 @@ ExternalProject_Add(fletch
     -DBUILD_SHARED_LIBS:BOOL=ON
     -Dfletch_BUILD_CXX11:BOOL=ON
     -Dfletch_BUILD_WITH_PYTHON:BOOL=${TELESCULPTOR_ENABLE_PYTHON}
-    -Dfletch_ENABLE_Boost:BOOL=ON
+    -Dfletch_ENABLE_Boost:BOOL=${TELESCULPTOR_ENABLE_FLETCH_BOOST}
     -DBoost_SELECT_VERSION:STRING=1.65.1
     -Dfletch_ENABLE_Caffe:BOOL=OFF
     -Dfletch_ENABLE_Caffe_Segnet:BOOL=OFF
-    -Dfletch_ENABLE_Ceres:BOOL=ON
+    -Dfletch_ENABLE_Ceres:BOOL=${TELESCULPTOR_ENABLE_FLETCH_CERES}
     -Dfletch_ENABLE_CppDB:BOOL=OFF
     -Dfletch_ENABLE_Darknet:BOOL=OFF
     -Dfletch_ENABLE_Darknet_OpenCV:BOOL=OFF
-    -Dfletch_ENABLE_Eigen:BOOL=ON
-    -Dfletch_ENABLE_FFmpeg:BOOL=ON
+    -Dfletch_ENABLE_Eigen:BOOL=${TELESCULPTOR_ENABLE_FLETCH_EIGEN}
+    -Dfletch_ENABLE_FFmpeg:BOOL=${TELESCULPTOR_ENABLE_FLETCH_FFMPEG}
     -DFFmpeg_SELECT_VERSION:STRING=3.3.3
-    -Dfletch_ENABLE_GDAL:BOOL=ON
+    -Dfletch_ENABLE_GDAL:BOOL=${TELESCULPTOR_ENABLE_FLETCH_GDAL}
     -Dfletch_ENABLE_GEOS:BOOL=ON
```

Comment:
I think we only need this to support building some other packages in Fletch.  GDAL maybe?  We don't use it directly.  We should confirm which fletch package needs GEOS and not build it if using a system package.

Reply:
It looks like GEOS is needed by GDAL, PDAL and PostGIS (the latter of which we don't use). BTW, PDAL has a hard requirement on *fletch's* GDAL, which seems suspect but means turning on PDAL but not GDAL won't work.

Do we need GDAL directly? Do we want to support using system GEOS but fletch's GDAL?

The easy way to answer such questions (which is where I was initially going here) is to just make everything a separate option and let the user deal with setting things appropriately in the face of any transitive dependencies, since I consider this whole thing somewhat "expert" options anyway. Slightly harder is to hide e.g. SuiteSparse if cerse-solver is not enabled. Less trivially harder is to actually enforce dependencies somehow, especially as those may change in fletch.

Reply:
I suppose you could have separate switches for everything, but I'd rather keep it simple.  In the case of SuiteSparse we have no code in TeleSculptor or KWIVER that use it directly and I think it is only a prerequisite for Ceres Solver.  So it seems to make sense for the Ceres Solver option to control both of them together.  For other packages that we don't use directly but have multiple packages that require them, this gets a little messy.  The right fix is to do that in Fletch with an option that tells Fletch to enable the packages which are dependencies of the things I've asked for.  I think Dave Stoup is working on that.

Reply:
That might work for SuiteSparse, glog and SQLite (although even if you use PROJ from fletch, you may have system SQLite). What about geos (which is also required by GDAL and PDAL)?

If we don't need GDAL directly, perhaps we can make GDAL and GEOS == PDAL? But then we're making these dependency chains all-or-nothing. (For similar reasons, I am ambivalent about fletch 'forcing' on dependencies. Maybe I need Qt and/or OpenCV, but have perfectly usable system PNG, jpeg, zlib, etc. Too, such an option would actually defeat the purpose of this branch, by causing TeleSculptor — which needs qtExtensions and VTK — to try to build Qt from fletch.)